src/routes/transactions/tests/routes.rs

```diff
@@ -0,0 +1,166 @@
+        .manage(Arc::new(mock_http_client) as Arc<dyn HttpClient>)
+}
+
+#[rocket::async_test]
+async fn post_confirmation_success() {
+    let mock_http_client = {
+        let mut mock_http_client = MockHttpClient::new();
+
+        let mut chain_request = Request::new(config_uri!("/v1/chains/{}/", 4));
+        chain_request.timeout(Duration::from_millis(chain_info_request_timeout()));
+        mock_http_client
+            .expect_get()
+            .times(1)
+            .with(eq(chain_request))
+            .returning(move |_| {
+                Ok(Response {
+                    status_code: 200,
+                    body: String::from(crate::tests::json::CHAIN_INFO_RINKEBY),
+                })
+            });
+
+        // CONFIRMATION REQUEST
+        let mut backend_request = Request::new(
+                "https://safe-transaction.rinkeby.staging.gnosisdev.com/api/v1/multisig-transactions/0x2e4af4b451a493470f38625c5f78f710f02303eb32780896cb55357c00d48faa/confirmations/"
+                .to_string(),
+        );
+        backend_request.body(Some(json!({"signature": "bd42f5c205b544cc6397c8c2e592ca4ade02b8681673cc8c555ff1777b002ee959c3cca243a77a2de1bbe1b61413342ac7d6416a31ec0ff31bb1029e921202ee1c"}).to_string()));
+
+        mock_http_client
+            .expect_post()
+            .times(1)
+            .with(eq(backend_request))
+            .return_once(move |_| {
+                Ok(Response {
+                    status_code: 201,
+                    body: String::new(),
+                })
+            });
+
+        // // TX DETAILS
+        let mut details_request =
+            Request::new(String::from("https://safe-transaction.rinkeby.staging.gnosisdev.com/api/v1/multisig-transactions/0x2e4af4b451a493470f38625c5f78f710f02303eb32780896cb55357c00d48faa/"));
+        details_request.timeout(Duration::from_millis(transaction_request_timeout()));
+
+        mock_http_client
+            .expect_get()
+            .times(1)
+            .with(eq(details_request))
+            .return_once(move |_| {
+                Ok(Response {
+                    status_code: 200,
+                    body: String::from(crate::tests::json::BACKEND_MULTISIG_TRANSFER_TX),
+                })
+            });
+
+        // safe info fetch for cancellations
+        let mut safe_request = Request::new(String::from("https://safe-transaction.rinkeby.staging.gnosisdev.com/api/v1/safes/0x1230B3d59858296A31053C1b8562Ecf89A2f888b/"));
+        safe_request.timeout(Duration::from_millis(safe_info_request_timeout()));
+
+        mock_http_client
+            .expect_get()
+            .times(1) // From FETCHING CANCELLATION AND FROM ENRICHING TX DETAILS
+            .with(eq(safe_request))
+            .returning(move |_| {
+                Ok(Response {
+                    body: String::from(crate::tests::json::SAFE_WITH_MODULES),
+                    status_code: 200,
+                })
+            });
+
+        // Cancellation tx
```

Comment:
@fmrsabino It is possible to remove the specific mocking for `Cancelation TX` and `Known Addresses` as we would be able to catch this we the last method. Maybe it is more appropriate to test this in the tests for `TxDetails` ?